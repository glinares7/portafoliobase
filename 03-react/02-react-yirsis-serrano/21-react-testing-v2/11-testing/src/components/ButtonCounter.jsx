import React from "react";

const ButtonCounter = ({ name, value = 1, action }) => {
  return (
    <button aria-label={name} onClick={() => action(value)}>
      {value}
    </button>
  );
};

export default ButtonCounter;
