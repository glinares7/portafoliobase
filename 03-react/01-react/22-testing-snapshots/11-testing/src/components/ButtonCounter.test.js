import { render, screen } from "@testing-library/react";
import ButtonCounter from "./ButtonCounter";
describe("<ButtonCounter/>", () => {
  const action = jest.fn();

  beforeEach(() => {
    render(<ButtonCounter value={3} name="aumentar" action={action} />);
  });

  it("crea un botton correctamente", () => {
    // render(<ButtonCounter value={3} name="aumentar" action={action} />);
    expect(screen.getByLabelText("aumentar")).toBeInTheDocument();
  });
  // it("crea un botton correctamente sin mandar value", () => {
  //   render(<ButtonCounter name="aumentar" action={action} />);
  //   expect(screen.getByLabelText("aumentar")).toBeInTheDocument();
  // });

  it("crea un botton correctamente", () => {
    // render(<ButtonCounter value={3} name="aumentar" action={action} />);
    expect(screen.getByLabelText("aumentar")).toMatchSnapshot();
  });
});
