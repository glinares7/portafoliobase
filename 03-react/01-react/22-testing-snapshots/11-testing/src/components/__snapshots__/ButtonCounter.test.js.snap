// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonCounter/> crea un botton correctamente 1`] = `
<button
  aria-label="aumentar"
>
  3
</button>
`;
